### DISCUSSION TITLE
missing gulpfile

### DISCUSSION THREAD

**Discussion by good-orbit** [holger1411/understrap#1534] [Q&A]
i am trying to set up dev environment with gulp and sass and child theme and the child and parent themes of understrap are missing gulpfile, help please.

**Comment by bacoords** [comment]
Hi, 

Understrap has moved away from using gulp in favor of custom build scripts. We recommend reviewing the documentation here: https://docs.understrap.com/#/understrap-child/npm

If you're still interested in using gulp, you can use an older release (pre 1.0), but there will be some newer features missing out. https://github.com/understrap/understrap-child/releases/tag/v0.6.0

**Comment by good-orbit** [comment]
Thank you!

**Comment by good-orbit** [comment]
Hi Brian,

I followed the docs and when I run npm run watch I get this: 
[nodemon] clean exit - waiting for changes before restart

**Reply by bacoords** [reply]
That's great. If you save any changes to a scss or js file, it should then compile all of your assets for you

**Comment by good-orbit** [comment]
no it’s not great, that is an error. I can’t do anything after I run npm run watch. 

[nodemon] clean exit - waiting for changes before restart

**Comment by good-orbit** [comment]
oh i see, sorry, maybe this will work. thanks again.

**Comment by OoMariannaoO** [comment]
Hi,

I wanted to use Understrap for building my WordPress theme. However, the newest version doesn't come with gulpfile and apparently, it's recommended not to use it at all. So here's my question: which script should I run and where should I apply my css? I've tried running 'npm run watch' and making changes to ./sass/theme/_theme.scss but nothing changes on my frontend. I am hosting my website on a xampp server. Thanks for any suggestions : )

**Reply by bacoords** [reply]
Hi we recommend double checking every step in the docs and then giving us the output of `node -v` in your console and also giving us any output from your console when you run `npm run watch` and attempt to save the file.

https://docs.understrap.com/#/understrap-child/npm

**Reply by JamesVanWaza** [reply]
@OoMariannaoO also if you are able to post a screenshot of the error, then I can also assist. 
1. Check which version of understrap and understrap child you are running. 
2. run `node -v`
3. run `npm run watch`

**Reply by OoMariannaoO** [reply]
Hi, thanks for getting back to me.
My node version is 16.14.0
Npm version is 8.5.0
As for understrap, I belive the version is 1.1.0

I've followed the documentation and tried to run npm run watch on the parent theme but this does nothing:
$ npm run watch

> understrap@1.1.0 watch
> npm-run-all --parallel watch-run-*


> understrap@1.1.0 watch-run-js
> nodemon --watch src/js/ --ext js --exec "npm-run-all js"


> understrap@1.1.0 watch-run-css
> nodemon --watch src/sass/ --ext scss --exec "npm-run-all css"

[nodemon] 2.0.15
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): src\sass\**\*   
[nodemon] watching extensions: scss
[nodemon] starting `npm-run-all css`        
[nodemon] 2.0.15
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): src\js\**\*
[nodemon] watching extensions: js
[nodemon] starting `npm-run-all js`

> understrap@1.1.0 css
> npm-run-all css-compile css-postcss css-minify


> understrap@1.1.0 js
> npm-run-all js-compile js-minify


> understrap@1.1.0 css-compile
> sass --style expanded --source-map --embed-sources --no-error-css --quiet src/sass/theme.scss:css/theme.css src/sass/custom-editor-style.scss:css/custom-editor-style.css


> understrap@1.1.0 js-compile
> rollup --config src/build/rollup.config.js --sourcemap


C:\xampp_2\htdocs\mariannaillu\wp-content\themes\test\src\js\bootstrap.js, C:\xampp_2\htdocs\mariannaillu\wp-content\themes\test\src\js\skip-link-focus-fix.js, C:\xampp_2\htdocs\mariannaillu\wp-content\themes\test\src\js\custom-javascript.js → js\theme.js...
created js\theme.js in 2.3s

> understrap@1.1.0 js-minify
> terser  --config-file src/build/terser.config.json --output js/theme.min.js js/theme.js


> understrap@1.1.0 css-postcss
> postcss --config src/build/postcss.config.js --replace "css/*.css" "!css/*.rtl*.css" "!css/*.min.css"

[nodemon] clean exit - waiting for changes before restart

The output gets updated when I make changes to  _theme_variables.scss but nothing changes if I go to localhost/my_site. I am using the latest version of xampp. I am really confused as to why it's not doing anything. 
On a similar note, I've also tried the child component and npm run watch doesn't work on it either. The only thing that seems to do anything is npm run watch-bs, but I don't want  to work on a child theme (also, if run in Microsoft Edge refreshing the site resets the styling to its default setting; it's ok in Chrome). 

I would appreciate any help in getting the parent theme to work.

**Reply by bacoords** [reply]
I've learned from others that some local environments, especially XAMPP, can be super aggressive in their caching. From what I can tell here, the theme assets are updating but the local site is caching the old CSS? Can you try keeping your dev tools open and doing a hard reload to see if the assets show?

Also- are you developing for Bootstrap 4 or 5? The parent theme doesn't compile the Bootstrap 4 assets in watch.
